src/redux/lobby/saga.ts

```diff
@@ -46,9 +46,24 @@ export default function* lobbySaga(address: string) {
 };
 
 // maps { '0xabc': challenge1Data, ... } to [challenge1Data, ....]
-const challengeTransformer = (dict) => Object.keys(dict.value).map((key) => dict.value[key]);
+const challengeTransformer = (dict) => {
+  return Object.keys(dict.value).map((key) => {
+    return dict.value[key];
+  }).filter((challenge) => {
+    return challenge.expiresAt > Date.now().toFixed()
```

Comment:
It might be worth adding a similar condition to the firebase query, to restrict the data you sync to those challenges that were active at the time you entered the lobby. It won't help with the accuracy but it would reduce the impact as the number of orphaned challenges increase.

Reply:
👍